Update operator, must-gather and podvm-builder ubi references

dbkreling opened the pull request:
Update ubi-minimal references from to **9.7-1763362218** and ubi to **9.7-1763340522** on the operator, must-gather and podvm-builder.

Also replace the "latest" tag on pccs.yaml.in by an epoch tag in order to improve stability, control and reprodicibility.

Issue: https://issues.redhat.com/browse/KATA-4311

dbkreling committed:
Update UBI references to 9.7 (podvm-builder)

This commit updates ubi reference from 9.6-1760340943 to 9.7-1764578509.

Signed-off-by: Daniel Kreling <dkreling@redhat.com>

update the RPM lockfile for UBI9.7

Signed-off-by: Julien Ropé <jrope@redhat.com>

dbkreling committed:
Update ubi-minimal to 9.7 (must-gather)

These will refresh the Health Index of those images.

Signed-off-by: Daniel Kreling <dkreling@redhat.com>
Signed-off-by: Julien Ropé <jrope@redhat.com>

littlejawa commented on must-gather/Dockerfile:
```diff
@@ -1,6 +1,6 @@
 FROM registry.redhat.io/openshift4/ose-must-gather-rhel9:v4.19 as builder
 
-FROM registry.access.redhat.com/ubi9/ubi-minimal:9.6-1760515502
+FROM registry.access.redhat.com/ubi10/ubi-minimal:10.1-1762189261
```
This tag is for registry.access.redhat.com/ubi10/ubi
ubi and ubi-minimal don't have the same tags. This is why the build is failing.

gkurz commented on Dockerfile:
```diff
@@ -29,7 +29,7 @@ RUN . ./controller-tools-ver && mv bin/controller-gen bin/controller-gen-$CONTRO
 
 # Use OpenShift base image
-FROM registry.access.redhat.com/ubi9/ubi-minimal:9.6-1760515502
+FROM registry.access.redhat.com/ubi10/ubi-minimal:10.1-1762189261
```
Since this is bumping the operator as well, maybe update the patch title to ` ... (must-gather, operator)` ?

gkurz replied:
@dbkreling, what's the motivation to bump the operator to ubi10 ? Asking because the jump from ubi9 to ubi10 appears to generate quite some extra work on the prow side. If it isn't strictly required I guess we should do this post release and simply bump to current latest [9.7-1763362218](https://catalog.redhat.com/en/software/containers/ubi9-minimal/61832888c0d15aff4912fe0d) to fix the health index.

gkurz replied:
I think this change should be dropped since we're downgrading it to 9.7 at the end.

littlejawa commented on Dockerfile:
```diff
@@ -29,7 +29,7 @@ RUN . ./controller-tools-ver && mv bin/controller-gen bin/controller-gen-$CONTRO
 
 # Use OpenShift base image
-FROM registry.access.redhat.com/ubi9/ubi-minimal:9.6-1760515502
+FROM registry.access.redhat.com/ubi10/ubi-minimal:10.1-1762189261
```
This tag is for registry.access.redhat.com/ubi10/ubi
ubi and ubi-minimal don't have the same tags. This is why the build is failing.

gkurz replied:
Yes, just realized that. Tag for ubi10 minimal should be `10.1-1763362715`

gkurz reviewed (dismissed):
With the suggested commit title fix *and* the right tag for ubi10 minimal :wink:

/lgtm

Thanks @dbkreling !

littlejawa commented:
You will need to add a RPM lockfile for the operator's build now that you install packages in it.
Also, do you really need the `redhat.repo` file for must-gather? It seems all the packages referenced in the lock file come from ubi repo... I think we don't need a subscription to build the must-gather actually.

dbkreling commented:
Last forced push removed the redhat.repo for must-gather and also added "rpms.in.yaml" and "rpms.lock.file".

wainersm commented:
Hi @dbkreling ! The ci/prow/* jobs are failing because they are still using the rhel-9 buildroot as you suspected.

The current base image is `base-rhel9` (see https://github.com/openshift/release/blob/master/ci-operator/config/openshift/sandboxed-containers-operator/openshift-sandboxed-containers-operator-devel.yaml#L5) and we will need to bump to `base-rhel10`. I don't know if `base-rhel10` is available in 4.19, otherwise we will need to bump the line https://github.com/openshift/release/blob/master/ci-operator/config/openshift/sandboxed-containers-operator/openshift-sandboxed-containers-operator-devel.yaml#L3 too.

I think above is enough, but more things to consider:

* It will also run `binary_build_commands` (line 6) in a container with `rhel-9-golang-1.24-openshift-4.20` (line 11). Do we want rhel-10 too?
* Likewise `GOFLAGS="" make test` in line 39.
* In line 20, instructs to replace `registry.redhat.io/openshift-sandboxed-containers/osc-rhel9-operator@sha256:.*` with the just built operator image in the bundle image. We need to check if this should be changed too.

gkurz commented on must-gather/Dockerfile:
```diff
@@ -1,6 +1,6 @@
 FROM registry.redhat.io/openshift4/ose-must-gather-rhel9:v4.19 as builder
 
-FROM registry.access.redhat.com/ubi9/ubi-minimal:9.6-1760515502
+FROM registry.access.redhat.com/ubi10/ubi-minimal:10.1-1762189261
```
Same question here. We certainly do want to move away from the B health index of [9.6-1760515502](https://catalog.redhat.com/en/software/containers/ubi9/ubi/615bcf606feffc5384e8452e?image=68ecb4d792068b299f0cf7ff)  but wouldn't it be safer to stick to ubi9 ?

gkurz replied:
I think this change should be dropped since we're downgrading it to 9.7 at the end. This should simplify the changes to the RPM lock file.

gkurz commented on Dockerfile:
```diff
@@ -1,4 +1,4 @@
-FROM registry.access.redhat.com/ubi9/go-toolset:1.24.6-1762373805 as builder
+FROM registry.access.redhat.com/ubi9/go-toolset:1.25-1763633888 as builder
```
Please use the `1.25.3-1763633888` tag, it is more informative.

gkurz commented on Dockerfile:
```diff
@@ -29,13 +29,12 @@ RUN . ./controller-tools-ver && mv bin/controller-gen bin/controller-gen-$CONTRO
 
 # Use OpenShift base image
-FROM registry.access.redhat.com/ubi10/ubi-minimal:10.1-1763362715
+FROM registry.access.redhat.com/ubi9/ubi-minimal:9.7-1763362218
```
The 10.1 bump is de facto not needed since we downgrade it in the same PR. For the sake of simplicity, can you please drop the change that does the 10.1 bump ?

gkurz commented:
> Downgrade non-critical references of ubi and ubi-minimal to 9.7
> Prow is not yet fully ready to handle ubi10 images and due to time constraints
> we should downgrade to ubi-9.7.
> 
> However, rhel-9 images should be upgraded to use the latest health index A of
> rhel-9.7 ones.
> 
> pccs.yaml.in 'latest' reference was changed due to future reproducibility.
> 
> Signed-off-by: Daniel Kreling <dkreling@redhat.com>
> Reviewed-by: Gregory Kurz <gkurz@redhat.com>
> 

We don't use `Reviewed-by:` tags in this repo. We use approval at the PR level in GH instead.
Also, this tag would only be added when I consider the commit to be ready, which isn't the case yet.
Please discard it.

gkurz commented on Dockerfile:
```diff
@@ -29,7 +29,7 @@ RUN . ./controller-tools-ver && mv bin/controller-gen bin/controller-gen-$CONTRO
 
 # Use OpenShift base image
-FROM registry.access.redhat.com/ubi10/ubi-minimal:10.1-1762189261
+FROM registry.access.redhat.com/ubi10/ubi-minimal:10.1-1763362715
```
For the same reason, this commit can be dropped since we're sticking to 9.7.

dbkreling committed:
Replace "latest" for epoch tag for PCCS

This will bring more stability, repdoducibility and control over
the base images used in these files.

Signed-off-by: Daniel Kreling <dkreling@redhat.com>
Signed-off-by: Julien Ropé <jrope@redhat.com>

dbkreling commented:
/retest

gkurz commented on config/peerpods/podvm/ubi.repo:
```diff
@@ -1,62 +1,62 @@
-[ubi-9-for-$basearch-baseos-rpms]
+[ubi-9-baseos-rpms]
```
I strongly suspect `for-$basearch-` is required for multi-arch support.

gkurz commented on config/peerpods/podvm/ubi.repo:
```diff
@@ -1,62 +1,62 @@
 name = Red Hat Universal Base Image 9 (Source RPMs) - BaseOS
 baseurl = https://cdn-ubi.redhat.com/content/public/ubi/dist/ubi9/9/$basearch/baseos/source/SRPMS
-enabled = 1
+enabled = 0
```
This got enabled by 154f467dd6e55. I think we should keep it.

gkurz commented on config/peerpods/podvm/ubi.repo:
```diff
@@ -1,62 +1,62 @@
 name = Red Hat Universal Base Image 9 (Source RPMs) - AppStream
 baseurl = https://cdn-ubi.redhat.com/content/public/ubi/dist/ubi9/9/$basearch/appstream/source/SRPMS
-enabled = 1
+enabled = 0
```
ditto

gkurz commented on config/peerpods/podvm/ubi.repo:
```diff
@@ -1,62 +1,62 @@
 name = Red Hat Universal Base Image 9 (Source RPMs) - CodeReady Builder
 baseurl = https://cdn-ubi.redhat.com/content/public/ubi/dist/ubi9/9/$basearch/codeready-builder/source/SRPMS
-enabled = 1
+enabled = 0
```
ditto

gkurz commented on must-gather/rpms.lock.yaml:
```diff
@@ -5,60 +5,36 @@ arches:
   packages:
   - url: https://cdn-ubi.redhat.com/content/public/ubi/dist/ubi9/9/s390x/baseos/os/Packages/r/rsync-3.2.5-3.el9.s390x.rpm
-    repoid: ubi-9-for-s390x-baseos-rpms
+    repoid: ubi-9-baseos-rpms
```
It looks like this might break multi-arch support and all the `for-....` should be kept in this PR.

gkurz commented on scripts/kata-install/Dockerfile:
```diff
@@ -1,4 +1,4 @@
-FROM registry.access.redhat.com/ubi9/skopeo:9.6-1760517870
+FROM registry.access.redhat.com/ubi9/skopeo:9.7-1763368371
```
Wrong commit.

littlejawa replied:
Yep, I think we should drop that commit - we don't use this Dockerfile in our builds. If needs an update, it probably belongs to a separate PR.

gkurz replied:
My comment was because it was hijacking a commit for another component. I'm personally fine to bump this in this PR, while here, as soon as it is in its own commit.

dbkreling committed:
Bump version of skopeo to 9.7

Updating it to move away from potential health indexes degradation
of version 9.6.

Signed-off-by: Daniel Kreling <dkreling@redhat.com>
Signed-off-by: Julien Ropé <jrope@redhat.com>

dbkreling committed:
Update Dockerfile ubi images

Go-toolset ubi should be bumped to refresh health index of the image.

The same applies to the ubi-minimal.

Signed-off-by: Daniel Kreling <dkreling@redhat.com>
Signed-off-by: Julien Ropé <jrope@redhat.com>

gkurz commented:
Postponing to OSC 1.11.z.

/hold

littlejawa commented:
I have updated the lock file for podvm-builder.
Running the lock file update for must gather didn't bring any change - the few packages we install apparently didn't change since 9.6.

gkurz commented:
> I have updated the lock file for podvm-builder.

Maybe squash this change into commit `Update UBI references to 9.7 (podvm-builder)` ?

littlejawa commented:
/unhold

gkurz reviewed (approved):
/lgtm

Thanks @littlejawa !
